hedera-node/hedera-evm-api/src/main/java/com/hedera/services/evm/store/models/HederaEvmAccount.java

```diff
@@ -15,9 +15,67 @@
  */
 package com.hedera.services.evm.store.models;
 
+import com.google.common.base.MoreObjects;
+import com.google.protobuf.ByteString;
+import com.hedera.services.utils.EthSigsUtils;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.tuweni.bytes.Bytes;
 import org.hyperledger.besu.datatypes.Address;
 
-public interface HederaEvmAccount {
+public class HederaEvmAccount {
 
-    Address canonicalAddress();
+    public static final ByteString ECDSA_KEY_ALIAS_PREFIX =
+            ByteString.copyFrom(new byte[] {0x3a, 0x21});
+    public static final int EVM_ADDRESS_SIZE = 20;
+    public static final int ECDSA_SECP256K1_ALIAS_SIZE = 35;
+
+    protected final Address address;
+    protected ByteString alias = ByteString.EMPTY;
+
+    public HederaEvmAccount(Address address) {
+        this.address = address;
+    }
+
+    public void setAlias(final ByteString alias) {
+        this.alias = alias;
+    }
+
+    public ByteString getAlias() {
+        return alias;
+    }
+
+    public Address canonicalAddress() {
+        if (alias.isEmpty()) {
+            return address;
+        } else {
+            if (alias.size() == EVM_ADDRESS_SIZE) {
+                return Address.wrap(Bytes.wrap(alias.toByteArray()));
+            } else if (alias.size() == ECDSA_SECP256K1_ALIAS_SIZE
+                    && alias.startsWith(ECDSA_KEY_ALIAS_PREFIX)) {
+                var addressBytes =
```

Comment:
Even using a native library, I think this calculation is (relatively) time-consuming. 

We could reduce the performance hit in a few ways, from least work (and least helpful) to most work (and most effective):
  1. Avoid recomputing per `HederaEvmAccount` instance by lazy-initializing a private field the first time someone calls `canonicalAddress`.
  2. Adding a LRU cache of some sort inside `EthSigsUtils.recoverAddressFromPubKey()` (though @OlegMazurov has demonstrated a Caffeine cache will not perform well).
  3. On creation of a `MerkleAccount` with a ECDSA alias, computing this value and storing it in state.

It's probably worth waiting for some perf results, but we might want to already start puzzling how (3) might work. 😇